receiver/dcgmreceiver/scraper.go

```diff
@@ -31,23 +31,45 @@ import (
 )
 
 type dcgmScraper struct {
-	config   *Config
-	settings receiver.CreateSettings
-	client   *dcgmClient
-	mb       *metadata.MetricsBuilder
+	config        *Config
+	settings      receiver.CreateSettings
+	client        *dcgmClient
+	mb            *metadata.MetricsBuilder
+	handleCleanup func()
 }
 
 func newDcgmScraper(config *Config, settings receiver.CreateSettings) *dcgmScraper {
 	return &dcgmScraper{config: config, settings: settings}
 }
 
-func (s *dcgmScraper) start(_ context.Context, _ component.Host) error {
-	var err error
-	s.client, err = newClient(s.config, s.settings.Logger)
+// initClient will try to create a new dcgmClient if currently has no client;
+// it will try to initialize the communication with the DCGM service; if
+// success, create a client; only return errors if DCGM service is available but
+// failed to create client.
+func (s *dcgmScraper) initClient() error {
+	if s.client != nil {
+		return nil
+	}
+	if !isDcgmInstalled() {
+		s.settings.Logger.Warn("can not initialize a DCGM client; DCGM is not installed.")
```

Comment:
Nit: cannot.
```suggestion
		s.settings.Logger.Warn("cannot initialize a DCGM client; DCGM is not installed.")
```
Also in the comment below.

Reply:
Done.